compiler/testData/codegen/box/ranges/forInRangeWithUpperBoundMinus1.kt

```diff
@@ -0,0 +1,208 @@
+// TARGET_BACKEND: JVM_IR
+// WITH_STDLIB
+
+fun box(): String {
+    noUnderflow()
+    testByteArray()
```

Comment:
That looks like overkill :). If you really want to go in that direction, it's better to check that "optimized" and "non-optimized" loop do the same thing (e.g., collect container elements to a string and check that resulting strings are equal). "Non-optimized" loop is basically the same loop with an explicitly allocated range, such as:
```
// optimized
for (x in 0..array.size-1) { ... }
// non-optimized
val range = 0..array.size-1
for (x in range) { ... }
```

Also, please bytecodeText tests for corresponding cases. Here we are targeting particular HotSpot loop optimization heuristics. It's rather complex to check that instructions are generated in proper order, BUT here's what I do in such case:
1. Check manually that generated counter loop does end with an increment (IINC). This is equivalent to 'for (...;...;++i) { ... }` in Java. Loop optimizations in HotSpot are looking for such code. Even if you have 100% equivalent loop that _doesn't end with an increment_, HotSpot doesn't treat it as a counter loop, corresponding optimizations are not triggered, and resutling bytecode is about 20% slower in rather simple cases.
2. Add a bytecodeText test with pattern like
```
// x IADD
// x ISUB
// x ILOAD
// x ISTORE
// x IINC
```
Usually changes in the number of generated IADD / ISUB / ILOAD / ISTORE / IINC instructions are enough to identify that bytecode shape might be wrong and needs manual check.

Reply:
Indeed, It is overkill , but considering that `for-loop` is a very fundamental part of a language, I don't want my kotlin journey stoped by any surprise in this one. However, I also completely agree to remove them now or later, please feel free to do that.

Reply:
I checked the bytecode of all test cases under `testData/codegen/bytecodeText/forLoop/forInRangeWithUpperBoundMinus1`, each `iinc` followed by a `goto`. It seems okay.